Examples/Scripts/Python/truth_tracking_chi2.py

```diff
@@ -0,0 +1,219 @@
+            EtaConfig(-2.0, 2.0),
+            ParticleConfig(2, acts.PdgParticle.eMuon, False),
+            multiplicity=1,
+            rnd=rnd,
+            outputDirRoot=outputDir,
+        )
+    else:
+        acts.logging.getLogger("Truth tracking example").info(
+            "Reading particles from %s", inputParticlePath.resolve()
+        )
+        assert inputParticlePath.exists()
+        s.addReader(
+            RootParticleReader(
+                level=acts.logging.INFO,
+                filePath=str(inputParticlePath.resolve()),
+                particleCollection="particles_input",
+                orderedEvents=False,
+            )
+        )
+
+    s = addFatras(
+        s,
+        trackingGeometry,
+        field,
+        rnd=rnd,
+    )
+
+    s = addDigitization(
+        s,
+        trackingGeometry,
+        field,
+        digiConfigFile=digiConfigFile,
+        rnd=rnd,
+    )
+
+    s = addSeeding(
+        s,
+        trackingGeometry,
+        field,
+        seedingAlgorithm=SeedingAlgorithm.TruthSmeared,
+        rnd=rnd,
+        truthSeedRanges=TruthSeedRanges(
+            pt=(500 * u.MeV, None),
+            nHits=(9, None),
+        ),
+    )
+
+    s = addChi2Tracks(
+        s,
+        trackingGeometry,
+        field,
+        nUpdates,
+        multipleScattering,
+        energyLoss,
+    )
+
+    # Output
+    s.addWriter(
+        acts.examples.RootTrajectoryStatesWriter(
+            level=acts.logging.INFO,
+            inputTrajectories="trajectories",
+            inputParticles="truth_seeds_selected",
+            inputSimHits="simhits",
+            inputMeasurementParticlesMap="measurement_particles_map",
+            inputMeasurementSimHitsMap="measurement_simhits_map",
+            filePath=str(outputDir / "trackstates_fitter.root"),
+        )
+    )
+
+    s.addWriter(
+        acts.examples.RootTrajectorySummaryWriter(
+            level=acts.logging.INFO,
+            inputTrajectories="trajectories",
+            inputParticles="truth_seeds_selected",
+            inputMeasurementParticlesMap="measurement_particles_map",
+            filePath=str(outputDir / "tracksummary_fitter.root"),
+        )
+    )
+
+    s.addWriter(
+        acts.examples.TrackFinderPerformanceWriter(
+            level=acts.logging.INFO,
+            inputProtoTracks="sortedprototracks" if directNavigation else "prototracks",
+            inputParticles="truth_seeds_selected",
+            inputMeasurementParticlesMap="measurement_particles_map",
+            filePath=str(outputDir / "performance_track_finder.root"),
+        )
+    )
+
+    s.addWriter(
+        acts.examples.TrackFitterPerformanceWriter(
+            level=acts.logging.INFO,
+            inputTrajectories="trajectories",
+            inputParticles="truth_seeds_selected",
+            inputMeasurementParticlesMap="measurement_particles_map",
+            filePath=str(outputDir / "performance_track_fitter.root"),
+        )
+    )
```

Comment:
can this be part of the `addChi2` method? similar to what we have in the `reconstruction.py` `add*` methods

Reply:
I would adapt it, when I'll merge it into `reconstruction.py`